code/_globalvars/lists/mobs.dm

```diff
@@ -34,3 +34,22 @@ GLOBAL_LIST_EMPTY(all_languages)
 GLOBAL_LIST_EMPTY(sentient_disease_instances)
 
 GLOBAL_LIST_EMPTY(latejoin_ai_cores)
+
+GLOBAL_LIST_EMPTY(mob_config_movespeed_type_lookup)
+
+/proc/initialize_mob_movespeed_type_lookup()
+	var/list/mob_typecache = typecacheof(/mob)
+	for(var/i in mob_typecache)
+		mob_typecache[i] = null
+	for(var/i in typecacheof(/mob/living/carbon/monkey))
+		mob_typecache[i] = "monkey"

```

Comment:
You missed the entire point of having the keyed list by using lookup values like this. The key should be a type and the value the number. You dont need to make a bunch of similar config value lookups for various types then, you can just have a single lookup of `movedelay_typecache[type]`

Reply:
partially pseudocode but you can have something like this
```
var/list/movedelay_typecache = list()
for(var/type_key in movedelay_config)
  var/delay_value = movedelay_config[type_key]
  for(var/mobtype in typesof(type_key))
    movedelay_typecache[mobtype] += delay_value
```

Reply:
oof ok